components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/api/impl/BusinessRulesApiServiceImpl.java

```diff
@@ -399,34 +400,61 @@ public Response loadDeploymentInfo(Request request, String businessRuleInstanceI
     }
 
     @Override
-    public Response redeployBusinessRule(Request request, String businessRuleInstanceID) throws NotFoundException {
+    public Response deployOrUndeployBusinessRule(Request request, String businessRuleInstanceID,
+                                                 boolean shouldUndeploy) throws NotFoundException {
         if (!hasPermission(request, RequestMethod.REDEPLOY_BUSINESS_RULE)) {
             return Response.status(Response.Status.UNAUTHORIZED).build();
         }
+
         TemplateManagerService templateManagerService = TemplateManagerInstance.getInstance();
         Gson gson = new GsonBuilder().disableHtmlEscaping().setPrettyPrinting().create();
         List<Object> responseData = new ArrayList<Object>();
         try {
-            int status = templateManagerService.redeployBusinessRule(businessRuleInstanceID);
+
+            status = templateManagerService.deployOrUndeployBusinessRule(businessRuleInstanceID, shouldUndeploy);
```

Comment:
where is this variable declared?

Reply:
it is already declared above. I will remove that and replace the same code here

Reply:
I fixed this issue